documentation/modules/exploit/linux/http/nagios_xi_configwizards_authenticated_rce.md

```diff
@@ -0,0 +1,108 @@
+## Vulnerable Application
+This module exploits CVE-2021-25926, CVE-2021-25927, and CVE-2021-25928: OS command injection vulnerabilities
+in `/nagiosxi/config/monitoringwizard.php` that enable an authenticated user to achieve remote code execution
+on NagiosXI from versions 5.5.6 to 5.7.5 as the `apache` user. There are three vulnerable configuration wizards
+(`windowswmi`, `switch`, `cloud-vm`) that use two vulnerable URL parameters
+(`ip_address`, `plugin_output_len`). The `windowswmi` configuration wizard uses the
+`plugin_output_len` parameter, while `switch` and `cloud-vm` use the `ip_address` parameter.
+
+The module's `check` method takes advantage of the `Msf::Exploit::Remote::HTTP::NagiosXi` mixin in order to authenticate
+to the target and obtain the version of Nagios XI installed, which is then used to check if the target is running a version
+of NagiosXI between versions 5.5.6 and 5.7.5.
+
+### Installation Steps
+Note: The module can attempt to complete the configuration steps after NagiosXI has been installed.
+
+#### Pre-installed OVA
+1. Download an OVA with NagiosXI installed
+    1. An OVA with a vulnerable NagiosXI 5.7.5 application can be downloaded from 
+    Nagios [here](https://assets.nagios.com/downloads/nagiosxi/5/ovf/nagiosxi-5.7.5-64.ova).
+    2. To download other vulnerable versions, replace the version number in the URL with the desired version.
+2. Configure the NagiosXI installation
+    1. The OVA has NagiosXI running on startup. Launch the VM and visit the VM's IP in a browser.
+    2. Configure the `nagiosadmin` user's password and whether you'd like to force HTTPS.
+    3. Login as the `nagiosadmin` user and accept the license agreement when prompted.
```

Comment:
Extra stuff I had to do:

1. Log in with username `root` and password `nagiosxi`.
2. Found I couldn't run `ipconfig` and the IP address wasn't showing up on the banner for some reason.
3. Had to change the OVA network from Bridged to NAT
4. Run `ip a` to get the IP address of the machine.
5. Go through defaults on the install. When you get to `Admin Account Settings` change the password from the generated one to `nagiosadmin`.

Think should mention some of these steps to help with troubleshooting potential errors.

Reply:
Yeah there's some troubleshooting errors I can add. Another weird one I kept having in the older versions was my nagiosxi time would be incorrect and I needed to manually set it. I'll make a note to either change the password or note the generated one also. No reason it has to be nagiosadmin, but you'll need to know the value for the module to work.

Reply:
Resolved with https://github.com/rapid7/metasploit-framework/pull/17494/commits/71554c57805613db963a7948d61dcbe7f6286f41